tests/Compiler/ComponentTagCompilerTest.php

```diff
@@ -35,4 +35,56 @@ public function testShouldCompileTwigConditions(): void
+    {
+        $compiler = new ComponentTagCompiler($component, 'alias');
+
+        $this->assertSame($expected, $compiler->compile());
+    }
+
+    /**
+     * @return array<string, string[]>
+     */
+    public function twigCommentsDataProvider(): array
+    {
+        return [
+            // component, expected
+            'comment in self-closing component' => [
+                '<Alert variable="{{value}}" {# This is comment #} variableWithoutQuotes={{value}} />',
+                '{% embed "@alias/alert.twig" with { props: {\'variable\': value,\'variableWithoutQuotes\': value} } %}{% endembed %}',
+            ],
+            'multiple comments in self-closing component' => [
+                '<Alert {# This is comment #} variable="{{value}}" {# This is second comment #} variableWithoutQuotes={{value}} />',
+                '{% embed "@alias/alert.twig" with { props: {\'variable\': value,\'variableWithoutQuotes\': value} } %}{% endembed %}',
+            ],
+            'comment in pair component' => [
+                '<Alert variable="{{value}}" {# This is comment #} variableWithoutQuotes={{value}}>Test</Alert>',
+                '{% embed "@alias/alert.twig" with { props: {\'variable\': value,\'variableWithoutQuotes\': value} } %}{% block content %}Test{% endblock %}{% endembed %}',
+            ],
+            'multiple comments in pair component' => [
+                '<Alert {# This is comment #} variable="{{value}}" {# This is second comment #} variableWithoutQuotes={{value}}>Test</Alert>',
+                '{% embed "@alias/alert.twig" with { props: {\'variable\': value,\'variableWithoutQuotes\': value} } %}{% block content %}Test{% endblock %}{% endembed %}',
+            ],
+            'comment out prop in self-closing component' => [
+                '<Alert {# variable="{{value}}" #} variableWithoutQuotes={{value}} />',
+                '{% embed "@alias/alert.twig" with { props: {\'variableWithoutQuotes\': value} } %}{% endembed %}',
+            ],
+            'comment out prop in pair component' => [
+                '<Alert {# variable="{{value}}" #} variableWithoutQuotes={{value}}>Test</Alert>',
+                '{% embed "@alias/alert.twig" with { props: {\'variableWithoutQuotes\': value} } %}{% block content %}Test{% endblock %}{% endembed %}',
+            ],
+            'empty comment in self-closing component' => [
+                '<Alert variable="{{value}}" {# #} variableWithoutQuotes={{value}} />',
+                '{% embed "@alias/alert.twig" with { props: {\'variable\': value,\'variableWithoutQuotes\': value} } %}{% endembed %}',
+            ],
+            'empty comment in pair component' => [
+                '<Alert variable="{{value}}" {#  #} variableWithoutQuotes={{value}}>Test</Alert>',
+                '{% embed "@alias/alert.twig" with { props: {\'variable\': value,\'variableWithoutQuotes\': value} } %}{% block content %}Test{% endblock %}{% endembed %}',
+            ],
```

Comment:
Maybe add also multiline comment?

```
<Alert variable="{{value}}" {# This is
comment
on multiple lines #} variableWithoutQuotes={{value}} />
```

Reply:
I do not know if this is relevant. Because, for example, we use

```twig
<Header
        activeMenuSection="{{ activeMenuSection }}"
        isCompaniesButtonVisible="{{ not hideCompaniesButton }}"
        isCzechLocale="{{ isCzechLocale }}"
        isFeedbackBadgeEnabled="{{ isFeedbackBadgeEnabled }}"
        isInverted="{{ isHeaderInverted }}"
        isLoginButtonVisible="{{ not hideLoginButton }}"
        isNavigationEnabled="{{ not disabledLinks }}"
        loginUrl="{{ loginUrl }}"
        logoutUrl="{{ logoutUrl }}"
        userName="{{ userName }}"
        userProfileUrl="{{ userProfileUrl }}"
        class="d-print-none"
/>
```

in our product templates and this is parsed anyway regardless of the new lines. E.g. I think that Twig flattens the template first, so any multi-line comment will get a single line.

Anyway, if I pass this structure directly into this test, it will not pass. Same as with a multi-line comment. But this structure is used without modification in this PR. So it leads to what I wrote above. That the template must by somehow flattened.